packages/terra-tabs/src/_CollapsibleTabs.jsx

```diff
@@ -73,6 +73,13 @@ class CollapsibleTabs extends React.Component {
     this.handleSelectionAnimation();
   }
 
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.children.length !== this.props.children.length) {
```

Comment:
I understand the intention of this enhancement is to handle the specific case of adding more tabs.

Some of these **dynamic** situations may already be handled but they're worth some thought and discussions to see if they should also be handled.

1. Reordering tabs. Length remains the same.
2. Changing the content of tab(s). Length remains the same.
3. Disabling / Enabling tab(s). Length remains the same.